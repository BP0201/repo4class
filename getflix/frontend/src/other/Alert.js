import React from 'react';

// used to create miniature alerts when an error occurs in a form
const Alert = ({ type = "danger", messages = [] }) => {
    for (let i = 0; i < messages.length; i++) {
        if (typeof messages[i] !== "string" || messages[i] === "") {
            messages.splice(i, 1)
        }
    }
    if (messages.length) {
        let messagesSet = new Set(messages)
        messages = Array.from(messagesSet)
        return (
            <div className={`alert alert-${type}`} role="alert">
                {messages.map((err, idx) => (
                    <p className='m-0 small' data-testid="alert-p" key={idx}>
                        {err}
                    </p>
                ))}
            </div>
        )
    }
}

export default Alert;