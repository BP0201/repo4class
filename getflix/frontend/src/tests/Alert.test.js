import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../other/Alert';

describe("Alert", () => {
  it("matches snapshot", () => {
    const { asFragment, getByText } = render(<Alert messages={["Hello", "This is a test"]} />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText("Hello")).toBeInTheDocument();
    expect(getByText("This is a test")).toBeInTheDocument();
  });

  it("should not render a duplicate message twice", () => {
    const { getAllByText } = render(<Alert messages={["Hello", "Hello"]} />);
    expect(getAllByText("Hello").length).toBe(1);
  });

  it("should not render a blank message", () => {
    const { getAllByTestId, getByText } = render(<Alert messages={["", "Hello", "", "Goodbye"]} />);
    expect(getByText("Hello")).toBeInTheDocument();
    expect(getByText("Goodbye")).toBeInTheDocument();
    // <p/> mapped from messages array containing message text
    expect(getAllByTestId("alert-p").length).toBe(2)
  });
});