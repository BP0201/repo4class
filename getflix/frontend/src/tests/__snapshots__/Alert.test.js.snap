// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="m-0 small"
      data-testid="alert-p"
    >
      Hello
    </p>
    <p
      class="m-0 small"
      data-testid="alert-p"
    >
      This is a test
    </p>
  </div>
</DocumentFragment>
`;
